### PR TITLE
scripts and tools: Fetch missing review comments in github-merge.py 

### PR DESCRIPTION
**Pull request by nkostoulas**
Use GitHub API pagination to do multiple requests if required.

Tested with some PRs that have a large number of comments.

For issue #15816

### CODE REVIEW

File: contrib/devtools/github-merge.py [hunk 1]
```diff
@@ -47,17 +47,41 @@ def git_config_get(option, default=None):
...
-        result = urlopen(req)
         reader = codecs.getreader('utf-8')
-        obj = json.load(reader(result))
+        if not use_pagination:
+            return json.load(reader(get_response(req_url, ghtoken)))
+
+        obj = []
+        page_num = 1
+        while True:
+            req_url_page = '%s?page=%d' % (req_url, page_num)
```
Comment (MarcoFalke) [inline contrib/devtools/github-merge.py]: Style-nit: Could use str.format

```suggestion
            req_url_page = '{}?page={}'.format(req_url, page_num)
```

File: contrib/devtools/github-merge.py [hunk 2]
```diff
@@ -47,17 +47,41 @@ def git_config_get(option, default=None):
...
+            result = get_response(req_url_page, ghtoken)
+            obj.extend(json.load(reader(result)))
+
+            link = result.headers.get('link', None)
+            if link is not None:
+                links = link.split(',')
```
Comment (MarcoFalke) [inline contrib/devtools/github-merge.py]: Could filter right here and get rid of the `found` bool?

```suggestion
                link_next = [l for l in link.split(',') if 'rel="next"' in l]
```

File: contrib/devtools/github-merge.py [hunk 3]
```diff
@@ -47,17 +47,41 @@ def git_config_get(option, default=None):
...
+                found = False
+                for link in links:
+                    if 'rel="next"' in link:
+                        page_num = int(link[link.find("page=") + 5])
```
Comment (MarcoFalke) [inline contrib/devtools/github-merge.py]: Doesn't work for double-digit page numbers, but fine

**Review by maflcko — APPROVED** [review]
utACK , feel free to ignore the comments

File: contrib/devtools/github-merge.py (hunk 2 above)
Comment (nkostoulas) [reply contrib/devtools/github-merge.py]: I rembered the bool was to avoid going through the whole list but I changed it anyway.

**Comment by MarcoFalke** [comment]
utACK 942ff2054b41fe3f78f1b3d88cfd032bc95fd62f

**Comment by laanwj** [comment]
utACK 942ff2054b41fe3f78f1b3d88cfd032bc95fd62f

I guess this makes it even more important to use an API token to not run into request limits.

**Pull request merged**